src/main/java/io/swagger/security/JwtUtil.java

```diff
@@ -0,0 +1,56 @@
+package io.swagger.security;
+
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.JWTVerifier;
+import com.auth0.jwt.algorithms.Algorithm;
+import com.auth0.jwt.exceptions.JWTCreationException;
+import com.auth0.jwt.exceptions.JWTVerificationException;
+import com.auth0.jwt.interfaces.DecodedJWT;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.stereotype.Component;
+
+import java.time.temporal.ChronoUnit;
+import java.util.Date;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Component
+public class JwtUtil {
+
+    private final String secret;
+    private final static String API_ISSUER = "MovieWatchlistAPI";
+
+    public JwtUtil(@Value("${app.security.jwtSecret}") String secret) {
+        this.secret = secret;
+    }
+
+    public String generateToken(Authentication authentication) throws IllegalArgumentException, JWTCreationException {
+        Date currentDate = new Date();
+        Date expiryDate = Date.from(currentDate.toInstant().plus(15, ChronoUnit.MINUTES));
+        String email = ((User) authentication.getPrincipal()).getUsername();
+        List<String> authorities = authentication.getAuthorities()
+                .stream()
+                .map(GrantedAuthority::getAuthority)
```

Comment:
Our app won't have any specific authorities, right? I recall that our "limited access" is solely based on simply authenticated and checking if a playlist is yours and/or publicly available.

Reply:
I only added two authorities, so yes we have extremely limited authorization policies. I was thinking it'd be good to secure our /users endpoint so regular users can't hit it.

- ROLE_USER
- ROLE_ADMIN

The ROLE_ADMIN is for us to hit the `/users` API

Anyone who registers with the API will get `ROLE_ADMIN`